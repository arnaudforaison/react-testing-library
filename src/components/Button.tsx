import { HTMLProps } from "react";

type Props = Omit<HTMLProps<HTMLButtonElement>, 'type'>;
const Button = ({ children, ...props }: Props) => (
  <button type="button" {...props}>{children}</button>
);
export default Button;
