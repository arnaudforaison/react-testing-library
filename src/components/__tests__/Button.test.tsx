import { render } from "@testing-library/react"
import Button from "../Button"

test('Button should render coucou text', () => {
    const {getByRole} = render(<Button>coucou</Button>);
    expect(getByRole('button')).toBeInTheDocument();
    expect(getByRole('button')).toHaveTextContent('coucou');
});

test('Button match snapshot', () => {
    const {asFragment} = render (<Button>Snapshot</Button>);
    expect(asFragment()).toMatchSnapshot();
})