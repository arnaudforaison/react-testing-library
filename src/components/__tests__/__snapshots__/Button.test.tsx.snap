// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button match snapshot 1`] = `
<DocumentFragment>
  <button
    type="button"
  >
    Snapshot
  </button>
</DocumentFragment>
`;
